src/Objects/Button.js

```diff
@@ -0,0 +1,31 @@
+import 'phaser';
+
+export default class Button extends Phaser.GameObjects.Container {
+    constructor(scene, x, y, key1, key2, text, targetScene) {
+        super(scene);
+        this.scene = scene;
+        this.x = x;
+        this.y = y;
+
+        this.button = this.scene.add.sprite(0, 0, key1).setInteractive();
+        this.text = this.scene.add.text(0, 0, text, { fontSize: '32px', fill: '#000' });
+        Phaser.Display.Align.In.Center(this.text, this.button);
+
+        this.add(this.button);
+        this.add(this.text);
+
+        this.button.on('pointerdown', function () {
+            this.scene.scene.start(targetScene);
+        }.bind(this));
```

Comment:
Fun fact with ES6 and functions. You can use the arrow function notation like this `() => this.scene.scene.start(targetScene)`, and `this` will be bound to where the function was declared, so you won't need to put `.bind(this)` after everything. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

Reply:
Nice, have updates every instance I found

Reply:
![image](https://user-images.githubusercontent.com/1319918/79641107-4f5ba780-818d-11ea-8107-3da0d7859155.png)